From: Greg Rose <gregory.v.rose@intel.com>
Date: Thu, 1 Jul 2010 13:39:23 +0000
Subject: igb: Add comment
References: FATE#311863 ,bnc#699089
Patch-mainline: v2.6.36-rc1
Git-commit: de42edde131cd09a556f0b90373569d64b92ef99

Add explanatory comment to avoid confusion when a pointer is set
to the second word of an array instead of the customary cast of a
pointer to the beginning of the array.

Signed-off-by: Greg Rose <gregory.v.rose@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/igb/igb_main.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/net/igb/igb_main.c
+++ b/drivers/net/igb/igb_main.c
@@ -4923,6 +4923,10 @@ static void igb_vf_reset_msg(struct igb_
 
 static int igb_set_vf_mac_addr(struct igb_adapter *adapter, u32 *msg, int vf)
 {
+	/*
+	 * The VF MAC Address is stored in a packed array of bytes
+	 * starting at the second 32 bit word of the msg array
+	 */
 	unsigned char *addr = (char *)&msg[1];
 	int err = -1;
 
